AstSemantics.md

```diff
@@ -47,15 +47,17 @@ Note that the local types `int32` and `int64` are not inherently signed or
 unsigned. The interpretation of these types is determined by individual
 operations.
 
-Parameters and local variables use local types.
+### Local Types
 
-### Expression Types
+*Local types* are a superset of the basic types, adding the following:
 
-*Expression types* include all the local types, and also:
+  * `funcptr`: a function identifier for use in `call_indirect`
 
-  * `void`: no value
```

Comment:
I'd like this document to explain why there's no void type. I'd also like to understand what a function does return if it doesn't return anything in source form.